core/auth/anonymous-session.ts

```diff
@@ -54,26 +67,23 @@ export const getAnonymousSession = async () => {
     throw new Error('AUTH_SECRET is not set');
   }
 
-  const session = await decode({
-    secret,
-    salt: `${cookiePrefix}${anonymousCookieName}`,
-    token: jwt.value,
-  });
+  try {
+    const session = await decode({
+      secret,
+      salt: `${cookiePrefix}${anonymousCookieName}`,
+      token: jwt.value,
+    });
 
-  return session;
-};
+    return session;
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error('Failed to decode anonymous session cookie, clearing invalid cookie', err);
 
-export const clearAnonymousSession = async () => {
-  const cookieJar = await cookies();
-  const useSecureCookies = await shouldUseSecureCookie();
-  const cookiePrefix = useSecureCookies ? '__Secure-' : '';
+    // Clear the invalid cookie
+    await clearAnonymousSession();
```

Comment:
Can't this entire PR be simplified if we simply omit this line and just return `null`, and allow the `anonymousSignIn()` to trigger in this condition? 

https://github.com/bigcommerce/catalyst/blob/365588846906d8d80a092955ceee9cb2cdf14f18/core/middlewares/with-auth.ts#L21-L24

We can rely on that condition because creating an anonymous session is independent of `req.auth`. When you create the anonymous session only sets the anonymous cookie and does not change the NextAuth session that drives `req.auth`. So if the jwt cannot be decoded and we simply return `null`, so long as the user is not actually signed in, it should trigger the logic above.

```suggestion

```